platforms/platforms.go

```diff
@@ -54,7 +56,11 @@ func DetermineBazelFilename(version string, includeSuffix bool) (string, error)
 		filenameSuffix = DetermineExecutableFilenameSuffix()
 	}
 
-	return fmt.Sprintf("bazel-%s-%s-%s%s", version, osName, machineName, filenameSuffix), nil
+	bazelFlavor := "bazel"
+	if val, ok := os.LookupEnv("BAZELISK_NOJDK"); ok && (val != "0") {
```

Comment:
This would be great to fall back to read it from the config as well with getEnvOrConfig : https://github.com/bazelbuild/bazelisk/blob/70694fe248380c6736a2947a52bcc9344dbfb365/core/core.go#L182

Reply:
done. but package `core` has to be splited to break import loop